--- a/targetallocator/config.go
+++ b/targetallocator/config.go
@@ -159,47 +159,47 @@
 	httpSD.HTTPClientConfig.FollowRedirects = false
 
 	httpSD.HTTPClientConfig.TLSConfig = commonconfig.TLSConfig{
-		InsecureSkipVerify: allocConf.TLSSetting.InsecureSkipVerify,
-		ServerName:         allocConf.TLSSetting.ServerName,
-		CAFile:             allocConf.TLSSetting.CAFile,
-		CertFile:           allocConf.TLSSetting.CertFile,
-		KeyFile:            allocConf.TLSSetting.KeyFile,
+		InsecureSkipVerify: allocConf.TLS.InsecureSkipVerify,
+		ServerName:         allocConf.TLS.ServerName,
+		CAFile:             allocConf.TLS.CAFile,
+		CertFile:           allocConf.TLS.CertFile,
+		KeyFile:            allocConf.TLS.KeyFile,
 	}
 
-	if allocConf.TLSSetting.CAPem != "" {
-		decodedCA, err := base64.StdEncoding.DecodeString(string(allocConf.TLSSetting.CAPem))
+	if allocConf.TLS.CAPem != "" {
+		decodedCA, err := base64.StdEncoding.DecodeString(string(allocConf.TLS.CAPem))
 		if err != nil {
 			return fmt.Errorf("failed to decode CA: %w", err)
 		}
 		httpSD.HTTPClientConfig.TLSConfig.CA = string(decodedCA)
 	}
 
-	if allocConf.TLSSetting.CertPem != "" {
-		decodedCert, err := base64.StdEncoding.DecodeString(string(allocConf.TLSSetting.CertPem))
+	if allocConf.TLS.CertPem != "" {
+		decodedCert, err := base64.StdEncoding.DecodeString(string(allocConf.TLS.CertPem))
 		if err != nil {
 			return fmt.Errorf("failed to decode Cert: %w", err)
 		}
 		httpSD.HTTPClientConfig.TLSConfig.Cert = string(decodedCert)
 	}
 
-	if allocConf.TLSSetting.KeyPem != "" {
-		decodedKey, err := base64.StdEncoding.DecodeString(string(allocConf.TLSSetting.KeyPem))
+	if allocConf.TLS.KeyPem != "" {
+		decodedKey, err := base64.StdEncoding.DecodeString(string(allocConf.TLS.KeyPem))
 		if err != nil {
 			return fmt.Errorf("failed to decode Key: %w", err)
 		}
 		httpSD.HTTPClientConfig.TLSConfig.Key = commonconfig.Secret(decodedKey)
 	}
 
-	if allocConf.TLSSetting.MinVersion != "" {
-		minVersion, err := convertTLSVersion(allocConf.TLSSetting.MinVersion)
+	if allocConf.TLS.MinVersion != "" {
+		minVersion, err := convertTLSVersion(allocConf.TLS.MinVersion)
 		if err != nil {
 			return err
 		}
 		httpSD.HTTPClientConfig.TLSConfig.MinVersion = minVersion
 	}
 
-	if allocConf.TLSSetting.MaxVersion != "" {
-		maxVersion, err := convertTLSVersion(allocConf.TLSSetting.MaxVersion)
+	if allocConf.TLS.MaxVersion != "" {
+		maxVersion, err := convertTLSVersion(allocConf.TLS.MaxVersion)
 		if err != nil {
 			return err
 		}
--- a/targetallocator/manager.go
+++ b/targetallocator/manager.go
@@ -130,9 +130,9 @@
 		}
 	}
 
-	addFile(m.cfg.TLSSetting.CAFile)
-	addFile(m.cfg.TLSSetting.CertFile)
-	addFile(m.cfg.TLSSetting.KeyFile)
+	addFile(m.cfg.TLS.CAFile)
+	addFile(m.cfg.TLS.CertFile)
+	addFile(m.cfg.TLS.KeyFile)
 
 	go func() {
 		for {
